test/ttmlir/Conversion/StableHLOToTTIR/conv2d_op.mlir

```diff
@@ -0,0 +1,23 @@
+// REQUIRES: stablehlo
+// RUN: ttmlir-opt --stablehlo-to-ttir-pipeline %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+module @jit_convolution attributes {} {
+  func.func public @test_convolution(%arg0: tensor<1x128x128x32xf32>, %arg1: tensor<64x32x3x3xf32>) -> tensor<1x128x128x64xf32> {
+    %0 = stablehlo.convolution(%arg0, %arg1)
+      dim_numbers = [b, 0, 1, f]x[o, i, 0, 1]->[b, 0, 1, f],
```

Comment:
I think `dim_numbers` should match tensor flow layout and read:
- `[b, 0, 1, f]x[0, 1, i, o]->[b, 0, 1, o]`

Let's not carry forward the TTNN style which mixes channels last activations with channels last weights.  We should file a bug on TTNN and fix it on their end too.

Reply:
In that case we have to make a pass in TTIR (or TTNN?) to do this. The only reason the tensorflow formatted weights don't cause an issue through forge is because forge inserts transposes on the weights which get constevalled before we move on to MLIR.